File: C# OOP/Inheritance/Exercises/Animals/Animal.cs
using System.Text;

namespace Animals
{
    class Animal
    {
        public string Name { get; set; }
        public int Age { get; set; }
        public string Gender { get; set; }

        public Animal(string name, int age, string gender)
        {
            this.Name = name;
            this.Age = age;
            this.Gender = gender;
        }

        public virtual void ProduceSound()
        {
        }

        private bool DataIsValid()
        {
            
        }

        public override string ToString()
        {
            StringBuilder sb = new StringBuilder();
            sb.AppendLine($"{GetType().Name}");
            sb.Append($"{this.Name} {this.Age} {this.Gender}");

            return sb.ToString().Trim();
        }
    }
}

File: Java Course/Java OOP/Inheritance/Exercises/Animals/src/Animals/Animal.java
package Animals;

public class Animal implements SoundProducable {
    private String name;
    private int age;
    private String gender;

    public Animal(String name, int age, String gender){
        this.name = name;
        this.age = age;
        this.gender = gender;
    }

    public String getName() {
        return name;
    }

    public int getAge() {
        return age;
    }

    public String getGender() {
        return gender;
    }

    @Override
    public String toString() {
        return this.getClass().getSimpleName() + "\r\n" +
                name + ' ' + age + ' ' + gender + "\r\n" +
                produceSound();
    }
}
